### PR TITLE
Only show attainable and owned special pets/mounts

### LINKED ISSUE #9498: Pawprints should not show for some Special Pets and Mounts
_edited by Alys 2018-11-11_

For the Special Pets and Mounts (those shown in the screenshot below), currently if you don't own them, you see pawprints for them. The pawprints are appropriate for only those Special Pets and Mounts that you are able to obtain by taking actions yourself (e.g., completing a Quest). The other Special Pets and Mounts are no longer available or are given out by staff at only certain times and so pawprints should not be shown for them (they should simply not appear in the list at all).

The Special Pets are configured here:
https://github.com/HabitRPG/habitica/blob/3265440bc44d6ca54cceb6cda4122ce981a0909f/website/common/script/content/stable.js#L54-L76
and the Special Mounts are:
https://github.com/HabitRPG/habitica/blob/3265440bc44d6ca54cceb6cda4122ce981a0909f/website/common/script/content/stable.js#L78-L94

They're added to the `petInfo` and `mountInfo` objects here:
https://github.com/HabitRPG/habitica/blob/3265440bc44d6ca54cceb6cda4122ce981a0909f/website/common/script/content/stable.js#L96-L110

Those `petInfo` and `mountInfo` objects could contain a `hidePawPrint: true` attribute for those Special Pets/Mounts that shouldn't have a pawprint showing. No extra config would be needed for the other Specials so that would be consistent with how the non-special pets/mounts are handled.

The ones that should NOT show pawprints are:

	Pets:
	  'Wolf-Veteran': 'veteranWolf',
	  'Wolf-Cerberus': 'cerberusPup',
	  'Turkey-Base': 'turkey',
	  'JackOLantern-Base': 'jackolantern',
	  'Tiger-Veteran': 'veteranTiger',
	  'Turkey-Gilded': 'gildedTurkey',
	  'Lion-Veteran': 'veteranLion',
	  'Gryphon-RoyalPurple': 'royalPurpleGryphon',
	  'JackOLantern-Ghost': 'ghostJackolantern',
	  'Orca-Base': 'orca',
	  'Bear-Veteran': 'veteranBear',
	  'Fox-Veteran': 'veteranFox',
	  'JackOLantern-Glow': 'glowJackolantern',
	  'Gryphon-Gryphatrice': 'gryphatrice',

	Mounts:
	  'LionCub-Ethereal': 'etherealLion',
	  'Turkey-Base': 'turkey',
	  'Orca-Base': 'orca',
	  'Gryphon-RoyalPurple': 'royalPurpleGryphon',
	  'JackOLantern-Base': 'jackolantern',
	  'Turkey-Gilded': 'gildedTurkey',
	  'JackOLantern-Ghost': 'ghostJackolantern',
	  'Gryphon-Gryphatrice': 'gryphatrice',

The others can all be obtained from Quests, or bought with Hourglasses, or obtained by subscribing or joining a Group Plan so we want them to show pawprints.

We can expect other special pets/mounts to be added in future; some will need to have the hidePawPrint attribute and others won't.

image

### PR DESCRIPTION
**Pull request by fmaximus**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #9498

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

Extend specialPets and specialMount with canFind attribute.
Only show special pets and mounts that can be obtained, or are owned.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: c4d64716-9f93-4356-a436-0df52dbf44d5

### CODE REVIEW

**Review by paglias — CHANGES_REQUESTED** [review]
The code is working but this result in an API change for the Content API (habitica.com/api/v4/content) that we cannot accept as it's a breaking change. In particular the `specialPets` and `specialMounts` objects would go from being maps of strings to maps of objects. 

Maybe we could instead a new key to the petInfo or mountInfo maps saying if the pet/mount can be shown or not? Additions of new properties aren't breaking changes

File: website/client/src/components/inventory/stable/index.vue [hunk 1]
```diff
@@ -706,24 +706,26 @@ export default {
...
-                return isOwned('mount', this, userItems);
-              },
-              isAllowedToFeed () {
-                return type === 'pet' && this.isOwned() && !this.mountOwned();
-              },
-              isHatchable () {
-                return false;
-              },
-            });
+            if (this.content[`${type}Info`][specialKey].canFind || isOwned('mount', this, userItems)) {
```
Comment (paglias) [inline website/client/src/components/inventory/stable/index.vue]: I think having the `if` here and not as a function in the animal item like `isOwned` a few lines below means that to get a new pet to show up a full refresh or page change is necessary. Why not add a new method below `canFind` that makes the check and is used with a `v-show` or `v-if` in the template?

**Comment by paglias** [comment]
I'm not able to test the correct behavior, added a special pet which is not findable to my test user items but it doesn't show up

Schermata 2020-05-20 alle 15 28 38

**Comment by fmaximus** [comment]
With latest commit, I have verified that adding an unfindable pet to a user in mongodb,
will show the new pet after a sync.

**Comment by paglias** [comment]
thanks @fmaximus ! noted towards your 3rd tier

**Pull request merged**